src/final2/subtests/InvalidCommandTest.java

```diff
@@ -0,0 +1,942 @@
+import test.runs.ErrorRun;
+import test.runs.NoOutputRun;
+import test.runs.Run;
+
+/**
+ * Tests the program for error messages with not well-formed or invalid commands.
+ * 
+ * @author Roman Langrehr
+ * @since 26.03.2015
+ *
+ */
+public class InvalidCommandTest extends LangtonSubtest {
+
+	private static final String[] ARGUMENT_COMMANDS = new String[] {
+			"create",
+			"field",
+			"position",
+			"direction",
+			"escape",
+			"move"
+	};
+
+	private static final String[] INVALID_ANT_NAMES = new String[] {
+			"ab",
+			"\00E4", // ae
+			"$",
+			"*"
+	};
+
+	private static final String[] INVALID_NUMBERS = new String[] {
+			"a",
+			"12t",
+			"three",
+			"12.5",
+			"$",
+			"9999999999999999999999999999999999999999999999999999999999999999",
+			"+0",
+			"+1"
+	};
+
+	private static final String[] NO_ARGUMENT_COMMANDS = new String[] {
+			"ant",
+			"print",
+			"quit"
+	};
+
+	private static final String[] OUT_OF_BOUNDS_COORDINATES_FOR_4X4 = new String[] {
+			"-1",
+			"4"
+	};
+
+	private static final String[] VALID_ARGUMENTS = new String[] {
+			"h,1,3",
+			"1,2",
+			"a",
+			"a",
+			"a",
+			"1"
+	};
+
+	/**
+	 * Asserts that the program avoids the creation of ants on blocked cells.
+	 */
+	@Test
+	public void createAntOnBlockedCellTest() {
+		inputFile = new String[] {
+				"0000",
+				"0*00",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0b00",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create a,2,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0A00",
+				"0000"
+		};
+		runs = new Run[] {
+				new NoOutputRun("move 1"),
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+	}
+
+	/**
+	 * Asserts that the program avoids the situation with two ants with the same name
+	 */
+	@Test
+	public void createExistingAntTest() {
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create a,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new NoOutputRun("create b,0,0"),
+				new ErrorRun("create b,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+	}
+
+	/**
+	 * Asserts that the program handles invalid ant names
+	 */
+	@Test
+	public void invalidAntNameTest() {
+		for (String invalidAntName : INVALID_ANT_NAMES) {
+			inputFile = new String[] {
+					"0000",

```

Comment:
Just use `TASK_SHEET_INPUT_FILE1` or any other of the predefined files from `LangtonTest`. (Another advantage: We already know the tested class can read in this file)